packages/rpc/src/client.ts

```diff
@@ -89,192 +134,192 @@ export class RPCClient extends EventEmitter {
 	) => boolean;
 
 	private endpoint = 'https://discord.com/api';
+
 	private readonly transport: Class<typeof transports[keyof typeof transports]>;
-	private readonly _expecting: Map<string, { resolve: (value: unknown) => void; reject: (reason?: Error) => void }>;
+
+	private readonly _expecting: Map<
+		string,
+		{ error: RPCError; reject(reason?: Error): void; resolve(value: unknown): void }
+	>;
+
 	private _connectPromise: Promise<unknown> | undefined;
-	private readonly _subscriptions: Map<string, ({ shortcut }: { shortcut: string }) => void>;
 
 	/**
-	 * @param {RPCClientOptions} [options] Options for the client.
+	 * @param options - Options for the client.
 	 * You must provide a transport
 	 */
-	public constructor(options: RPCClientOptions = {}) {
+	public constructor(public readonly options: RPCClientOptions) {
 		super();
 
 		this.options = options;
 
-		this.accessToken = null;
-		this.clientId = null;
-
-		/**
-		 * Application used in this client
-		 * @type {?ClientApplication}
-		 */
-		this.application = null;
-
-		/**
-		 * User used in this application
-		 * @type {?Partial<APIUser>}
-		 */
-		this.user = null;
-
-		const Transport = options.transport ? transports[options.transport] : false;
-		if (!Transport) {
-			throw new TypeError('RPC_INVALID_TRANSPORT', options.transport as undefined);
+		const transport = transports[options.transport];
+		if (!transport) {
+			const typeError = new TypeError('INVALID_TRANSPORT', options.transport as any);
```

Comment:
Would be nice to have a better naming for this variable, maybe `invalidTransport`?